OA8TV3/DownloadManager/Domain/ExtensionRule.cs

```diff
@@ -0,0 +1,41 @@
+using System.Text.Json.Serialization;
+
+namespace DownloadManager.Domain
+{
+    public record ExtensionRule : AbstractRule, IComparable<AbstractRule>
+    {
+        [JsonPropertyName("extension")]
+        public required string Extension { get; init; }
+
+        public virtual bool Equals(ExtensionRule? other)
+        {
+            if (ReferenceEquals(null, other))
+            {
+                return false;
+            }
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+            return String.Equals(Extension, other.Extension, StringComparison.CurrentCultureIgnoreCase);
+        }
+
+        public override int CompareTo(AbstractRule? other)
```

Comment:
No, ennek virtual-nak kellett volna lennie, vagy absztraktnak és a származott osztálynak felülírnia. Így az ExtensionRule-ba tartozó összehasonlítás ott lenne implementálva, a pattern-es megoldás meg ott. VAGY ha mindenképpen kell tudnod a teljes hierarchiáról, akkor inkább egy külön `IComparer<T>` interfészt implementáltam volna. https://webmaster442.github.io/ultimatedotnetcheatsheet/#icomparert. A gond ezzel az, hogy ez az osztály innentől tud a teljes gyerek listájáról ami így sérti a single responsibility elvet és az open/closed-ot, mert minden esetben, amikor új gyerek jön a szülőt is módosítani kell. A compareres megoldás legalább csak az open/closed-ot sértené.